anndata/_core/views.py

```diff
@@ -69,7 +69,9 @@ def __new__(
         input_array: Sequence[Any],
         view_args: Tuple["anndata.AnnData", str, Tuple[str, ...]] = None,
     ):
-        arr = np.asarray(input_array).view(cls)
+        if not isinstance(input_array, np.ndarray):
+            input_array = np.array(input_array)
+        arr = input_array.view(cls)
```

Comment:
```suggestion
        arr = np.asanyarray(input_array).view(cls)
```

Could this work instead? I think it should if your class is literally a np.ndarray subclass.

Reply:
Thanks for the suggestion! I've added a test and it works.